### PR TITLE
FS-1259; Basic pushgateway metrics collection

### PR DESCRIPTION
**Pull request by jakeschuurmans**
#### What does this PR do

This allows fleet-scheduler to report go metrics to a pushgateway

### CODE REVIEW

File: cmd/inventory.go [hunk 1]
```diff
@@ -53,6 +49,19 @@ func inventory(ctx context.Context) error {
...
+	metricsPusher.AddCollector(collectors.NewGoCollector())
+	err = metricsPusher.Start()
+	if err != nil {
+		return err
+	}
+
+	otelCtx, otelShutdown := otelinit.InitOpenTelemetry(ctx, "fleet-scheduler")
+	defer otelShutdown(ctx)
+
+	otelCtxWithCancel, cancelFunc := context.WithCancel(otelCtx)
```
Comment (DoctorVin) [inline cmd/inventory.go:62]: Why do you need this second context? Isn't `otelCtx` a context with a cancelation function (specifically `otelShutdown`)?

File: internal/metrics/metrics.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+	mp.wg.Wait()
+}
+
+// Pushes metrics to the pushgateway
+func (mp *Pusher) Push() error {
+	mp.mu.Lock()
+	defer mp.mu.Unlock()
+
+	return mp.pusher.Push()
```
Comment (DoctorVin) [inline internal/metrics/metrics.go:101]: Where do we define what actually gets pushed here?

**Review by DoctorVin — COMMENTED** [review]
I feel like this has a very distinct C sound. It might have been more go-natural to work with channels instead. Using the mutex directly works though.

File: internal/metrics/metrics.go (hunk 2 above)
Comment (jakeschuurmans) [reply internal/metrics/metrics.go:101]: The collector. So i made it a bit "dynamic". Maybe a bit too dynamic. You can see on cmd/inventory.go:53 I add the Go Collector. That is what is being pushed. IT collects all the default go data about memstats, gc, and cpu.

File: cmd/inventory.go (hunk 1 above)
Comment (jakeschuurmans) [reply cmd/inventory.go:62]: Maybe I dont understand enough about context, but I cant find anything within otelinit that cancels the ctx, only shuts down opentelemetry.

Comment (DoctorVin) [reply cmd/inventory.go:62]: That `otelShutdown` function is [bogus](https://github.com/equinix-labs/otel-init-go/blob/14a1d0e47a1b1228f2ca3572a7421bd94e9ff595/otelinit/public.go#L36).

File: internal/metrics/metrics.go (hunk 2 above)
Comment (DoctorVin) [reply internal/metrics/metrics.go:101]: What are you your plans for app-specific metrics?

Comment (jakeschuurmans) [reply internal/metrics/metrics.go:101]: Tasks completed, tracking each type of task individually as we add more

For inventory task specifically I think the only thing I could record is condition creation counts.

Comment (jakeschuurmans) [reply internal/metrics/metrics.go:101]: Also, thinking about a more go like way to do the pusher thread. I dont like any of my options. Using the ctx seems the normal go way to do things, problem is then its not self contained. I would have to make changes to how we handle cancel ctx and stuff. I feel that the C way is the way (it always has been).

**Review by DoctorVin — APPROVED** [review]

**Pull request merged**